manifests/deploy.pp

```diff
@@ -27,10 +27,15 @@
   $servergroups = hiera('jboss::deploy::servergroups', undef),
   $controller   = $::jboss::controller,
   $runasdomain  = $::jboss::runasdomain,
+  $runtime_name = undef,
 ) {
   include jboss
   include jboss::internal::runtime::node
 
+  if $runtime_name != undef {
+    validate_re($runtime_name, '.+.(\.ear|\.zip|\.war|\.jar)$')
```

Comment:
!fix Should be `.+(\.ear|\.zip|\.war|\.jar)$`. Also add a fail message for end-users